physutils/tasks.py

```diff
@@ -0,0 +1,85 @@
+import logging
+from functools import wraps
+
+from bids import BIDSLayout
+from loguru import logger
+
+from physutils.io import load_from_bids, load_physio
+from physutils.physio import Physio
+
+LGR = logging.getLogger(__name__)
+LGR.setLevel(logging.DEBUG)
+
+try:
+    import pydra
+
+    pydra_imported = True
+except ImportError:
+    pydra_imported = False
+
+
+def mark_task(pydra_imported=pydra_imported):
+    def decorator(func):
+        if pydra_imported:
+            # If the decorator exists, apply it
+            @wraps(func)
+            def wrapped_func(*args, **kwargs):
+                logger.debug(f"Creating pydra task for {func.__name__}")
+                return pydra.mark.task(func)(*args, **kwargs)
+
+            return wrapped_func
+        # Otherwise, return the original function
+        return func
+
+    return decorator
```

Comment:
That may work, but I honestly was thinking about something like:

```
def task(func):
    @functools.wraps(func)
    def wrapper(*args, **kwargs):
        return func(*args, **kwargs)
    return wrapper
```
In a `utils` module, and then here:.
```
try:
    from pydra.mark import task
else:
    from .utils import task
```